### ISSUE TITLE
Fail to setup CloudFoundry on AWS region eu-central-1 with BOSH version 3232

### ISSUE THREAD

**Issue by djvdorp** [cloudfoundry/bosh#1298]
We are currently trying to deploy CloudFoundry on AWS (region: eu-central-1) as [per their instructions](https://docs.cloudfoundry.org/deploying/aws/index.html). At [the point where we run the command](https://docs.cloudfoundry.org/deploying/aws/setup_aws.html) `bosh aws create` we get the following error:

```
[ec2-user@ip-XXX-XXX-XXX-XXX cf]$ bosh aws create                                                                                                                                                                
Executing migration CreateKeyPairs
allocating 1 KeyPair(s)
Executing migration CreateVpc
creating internet gateway
creating security groups: open, bosh, bat, cf, web
creating subnets: bosh1, bosh_rds1, cf_elb1, cf_rds1, services_rds1, cf1, services1, bosh_rds2, cf_elb2, cf_rds2, services_rds2
  Making subnet bosh1 10.10.0.0/24:
  Making subnet bosh_rds1 10.10.1.0/24:
  Making subnet cf_elb1 10.10.2.0/24:
  Making subnet cf_rds1 10.10.3.0/24:
  Making subnet services_rds1 10.10.8.0/21:
  Making subnet cf1 10.10.16.0/20:
  Making subnet services1 10.10.32.0/20:
  Making subnet bosh_rds2 10.10.65.0/24:
  Making subnet cf_elb2 10.10.66.0/24:
  Making subnet cf_rds2 10.10.67.0/24:
  Making subnet services_rds2 10.10.72.0/21:
details in S3 receipt: aws_vpc_receipt and file: aws_vpc_receipt.yml

expected string value for option image_id.
```

The actual error, `expected string value for option image_id.`, seems to come from the `aws-sdk` (`aws-sdk-v1-1.60.2`) ruby gem, in file `lib/aws/core/option_grammar.rb` and function `apply_string_descriptor`. Is this because of the region eu-central-1, or is the `bosh_cli_plugin_aws` broken, or is this something else?
We see that the file `receipts/aws_vpc_receipt.yml` is actually put in the s3 bucket and it has file contents. Also there is `aws_migrations/migrations.yaml` in the s3 bucket.

We (obviously) tried finding the cause of this error myself, then stumbled upon a [Github issue](https://github.com/cloudfoundry/bosh/issues/722#issuecomment-152944860) stating that "`bosh aws *` commands have been deprecated few months ago. (written on Nov 2, 2015)", but is this the actual case, and what is the replacement to install Cloud Foundry on an AWS Environment then?

Relevant version information:
BOSH version: `1.3232.0`
Ruby version: `2.0.0p648`
Gemfile.lock: see attached [Gemfile.lock.txt](https://github.com/cloudfoundry/bosh/files/311997/Gemfile.lock.txt)

**Comment by cppforlife** [comment]
cc @Amit-PivotalLabs

**Comment by Amit-PivotalLabs** [comment]
Hey @djvdorp 

Not sure if you saw my response to your Slack question: https://cloudfoundry.slack.com/archives/general/p1465834173000016

The docs are in an intermediate state essentially waiting on new tooling, and the new tooling is in progress, but we can guide you in using the new tooling if you're willing to try it out.  It intentionally doesn't do everything the old "bosh aws bootstrap" tools did, for instance it doesn't create RDS instances (and assumes you'll deploy CF with the built-in postgres job, or manage RDS or some other external database yourself).  If you're interested in this guidance, let me know.

**Comment by djvdorp** [comment]
Hi @Amit-PivotalLabs 

I must apologize that I missed your response to my Slack question, which I will put here below for other people's reference which might be useful:

```
amitgupta [6:09 PM] 
@djvdorp: the problem with those docs and tools is they can get out of date. We're working on some tooling that will be actively supported. For now you've got a few options

1. Still try to use the "bosh aws" tooling. Last time we tried those instructions they worked, with some small road bumps though
2. Those docs also outline a manual procedure you could try
3. The new tooling is still in experimental phase, but I can point you to it
```

I did not have the opportunity yet to discuss this with my colleagues who are working with me on deploying CloudFoundry on AWS, but today we have spent quite some time on getting `option 1` to work. For tomorrow we were thinking about looking into `option 2`, but I would be very interested to learn more about `option 3` since I highly prefer automation over manual work. 

Could you point us towards information regarding this option, and guide us in using this new tooling to try out? Thank you for your time and effort so far, it is highly appreciated!

**Comment by Amit-PivotalLabs** [comment]
We'd like to remove the "bosh aws" from the documentation as soon as possible, so if you're willing to try option 3 the feedback would be very valuable for us, and should get us to a place where we can replace those docs with something that works sooner.

Option 3 is currently experimental and lives in the pivotal-cf-experimental org, but we'll be moving it over to the cloudfoundry-incubator org soon: https://github.com/pivotal-cf-experimental/bosh-bootloader.  If you open an issue there we can describe a workflow that should get you started with CF (later on, a polished set of docs will describe the workflow), though we're waiting on one feature from BOSH that's just around the corner, to support "BOSH 1.0" style manifests, since the existing tooling for generating CF manifests only supports the 1.0 style.

**Comment by dpb587-pivotal** [comment]
@djvdorp, I think you have been able to get CloudFoundry deployed now with some help from bbl, so I'm going to close this. Feel free to reopen if that's not the case, or start a new issue for other problems.
